lightning/src/util/ser_macros.rs

```diff
@@ -120,15 +126,17 @@ macro_rules! check_missing_tlv {
 	}};
 }
 
+/// Decode a TLV.
```

Comment:
Seems strange to export this - it doesn't really do anything by itself, it just writes a single field. I think the more useful things to export are `encode_tlv_stream` and `decode_tlv_stream`. That would require exporting this, but I think we should `#[doc(hidden)]` this and prefix it with a `_` to indicate that it isn't, itself, a stable API or something users should use directly.

Reply:
Yes sorry this needs to be re-worked.

Reply:
I removed the serialization commit for now, need more time to work on that.